### PR TITLE
feat: ingestion implementation v2

### PR DESCRIPTION
**Pull request by dengmingtong**
----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

## Summary

Implement Ingestion server v2 including below changes:
1, Support Fargate and EC2 in V2
2, Separate ECS cluster and other resources(alb, aga) to two nest stack
3, Support protocol, endpoint path, auth/without auth, aga enable/disable, domain, certificate change.
4, Add unit test cases to cover all change

## Implementation highlights

(describe how the merge request does for feature changes, share the RFC link if it has)

## Test checklist

- [x] add new test cases
- [x] all code changes are covered by unit tests
- [x] end-to-end tests
  - [ ] deploy web console with CloudFront + S3 + API gateway
  - [ ] deploy web console within VPC
  - [x] deploy ingestion server
    - [x] with MSK sink
    - [x] with KDS sink
    - [x] with S3 sink
  - [ ] deploy data processing
  - [ ] deploy data modeling
    - [ ] new Redshift Serverless
    - [ ] provisioned Redshift
    - [ ] Athena
  - [ ] deploy with reporting

## Is it a breaking change

- [ ] add parameters without default value in stack
- [ ] introduce new service permission in stack
- [ ] introduce new top level stack module

## Miscellaneous

- [ ] introduce new symbol link source file(s) to be shared among infra code, web console frontend, and web console backend

### CODE REVIEW

File: src/ingestion-server/custom-resource/update-alb-rules-v2/index.ts [hunk 1]
```diff
@@ -0,0 +1,809 @@
+/**
```
Comment (zxkane) [inline src/ingestion-server/custom-resource/update-alb-rules-v2/index.ts:1]: What's the difference compared to the original `update-alb-rules`?

File: src/ingestion-server/server-v2/condition-v2.ts [hunk 2]
```diff
@@ -14,42 +14,84 @@
 import { CfnCondition, CfnParameter, Fn } from 'aws-cdk-lib';
 import { Stream } from 'aws-cdk-lib/aws-kinesis';
 import { Construct } from 'constructs';
-import { SINK_TYPE_MODE } from '../../common/model';
+import { SINK_TYPE_MODE, ECS_INFRA_TYPE_MODE } from '../../common/model';
```
Comment (zxkane) [inline src/ingestion-server/server-v2/condition-v2.ts:18]: it's unnecessary to add `v2` in file name if it's already named `server-v2`

File: src/ingestion-server/server-v2/condition-v2.ts [hunk 3]
```diff
@@ -14,42 +14,84 @@
...
-  onDemandStackStream: Stream;
-  provisionedStackCondition: CfnCondition;
-  onDemandStackCondition: CfnCondition;
-}) {
-  const kinesisConditionsAndProps = [
+export function createECSTypeCondition(
+  scope: Construct,
+  ecsInfraType: string,
+) {
+  const ecsEC2Condition = new CfnCondition(scope, 'ecsEC2Condition', {
```
Comment (zxkane) [inline src/ingestion-server/server-v2/condition-v2.ts:44]: What's the primary difference between the IaC code for Ec2 and Fargate?

File: src/ingestion-server/server-v2/condition-v2.ts [hunk 4]
```diff
@@ -14,42 +14,84 @@
...
+      condition: ecsFargateCondition,
+      name: 'F',
+      ecsInfraType: ECS_INFRA_TYPE_MODE.FARGATE,
     },
-
   ];
+  return ecsInfraConditions;
+}
+
+export function createKinesisConditionsV2(
```
Comment (zxkane) [inline src/ingestion-server/server-v2/condition-v2.ts:65]: It's no difference whether KDS is provisioned or on demand.

File: src/ingestion-server/server-v2/private/alb-v2.ts [hunk 5]
```diff
@@ -67,9 +55,9 @@ export interface ApplicationLoadBalancerProps {
   certificateArn: string;
   domainName: string;
   sg: SecurityGroup;
-  service: BaseService;
+  // service: BaseService;
```
Comment (zxkane) [inline src/ingestion-server/server-v2/private/alb-v2.ts:58]: remove commeted code

File: src/ingestion-server/server-v2/private/ecs-ec2-cluster.ts [hunk 6]
```diff
@@ -0,0 +1,191 @@
...
+): EcsClusterResult {
+  const vpc = props.vpc;
+
+  const ecsCluster = new Cluster(scope, `${RESOURCE_ID_PREFIX}ecs-cluster`, {
+    vpc,
+    containerInsights: true,
+  });
+
+  const ecsAsgSetting = props.ec2FleetProps;
+  const isArm = ecsAsgSetting.isArm;
```
Comment (zxkane) [inline src/ingestion-server/server-v2/private/ecs-ec2-cluster.ts:86]: use arch instead

**Review by zxkane — CHANGES_REQUESTED** [review]

File: src/ingestion-server-v2-stack.ts [hunk 7]
```diff
@@ -32,30 +33,85 @@ import { OUTPUT_INGESTION_SERVER_DNS_SUFFIX, OUTPUT_INGESTION_SERVER_URL_SUFFIX
...
+  readonly workerStopTimeout: number;
+
+  readonly serverEndpointPath: string;
+  readonly serverCorsOrigin: string;
+  readonly protocol: string;
+  readonly domainName: string;
+  readonly certificateArn: string;
+  readonly notificationsTopicArn?: string;
+  readonly enableApplicationLoadBalancerAccessLog?: string;
+  readonly logBucketName?: string;
```
[the remark this answers is not in the document]
Comment (dengmingtong) [reply src/ingestion-server-v2-stack.ts:76]: Done

**Comment by zxkane** [comment]
### SonarQube Quality Gate Result
- **Result**: :exclamation: Error
- Triggered by @dengmingtong on `pull_request`

| Metric | Status | Value | Error Threshold |
|:------:|:------:|:-----:|:---------------:|
|Reliability rating|:white_check_mark: OK|1|> 1|
|Security rating|:white_check_mark: OK|1|> 1|
|Sqale rating|:white_check_mark: OK|1|> 1|
|Coverage|:white_check_mark: OK|83.60|< 80|
|Duplicated lines density|:white_check_mark: OK|5.20|> 30|
|Blocker violations|:white_check_mark: OK|0|> 0|
|Bugs|:white_check_mark: OK|0|> 0|
|Code smells|:white_check_mark: OK|13|> 40|
|Critical violations|:white_check_mark: OK|0|> 0|
|Major violations|:exclamation: Error|4|> 0|
|Security hotspots reviewed|:exclamation: Error|0|< 100|
|Vulnerabilities|:white_check_mark: OK|0|> 0|

[View on SonarQube](http://sonarqube:9000/dashboard?id=pr-896)
###### _updated: 6/16/2024, 15:46:27 (UTC+0)_

**Comment by zxkane** [comment]
# SonarQube Code Analytics
## Quality Gate failed

failed

## Additional information
*The following metrics might not affect the Quality Gate status but improving them will improve your project code quality.*

## Issues
bug   [0 Bugs](http://sonarqube:9000/project/issues?id=pr-896&resolved=false&sinceLeakPeriod=true&types=BUG)

vulnerability   [0 Vulnerabilities](http://sonarqube:9000/project/issues?id=pr-896&resolved=false&sinceLeakPeriod=true&types=VULNERABILITY)

code_smell   [0 Code Smells](http://sonarqube:9000/project/issues?id=pr-896&resolved=false&sinceLeakPeriod=true&types=CODE_SMELL)

## Coverage and Duplications
*No data* **Coverage**

*No data* **Duplication**

**Comment by zxkane** [comment]
### SonarQube Quality Gate Result
- **Result**: :exclamation: Error
- Triggered by @dengmingtong on `pull_request`

| Metric | Status | Value | Error Threshold |
|:------:|:------:|:-----:|:---------------:|
|Reliability rating|:white_check_mark: OK|1|> 1|
|Security rating|:white_check_mark: OK|1|> 1|
|Sqale rating|:white_check_mark: OK|1|> 1|
|Coverage|:white_check_mark: OK|83.70|< 80|
|Duplicated lines density|:white_check_mark: OK|5.20|> 30|
|Blocker violations|:white_check_mark: OK|0|> 0|
|Bugs|:white_check_mark: OK|0|> 0|
|Code smells|:white_check_mark: OK|13|> 40|
|Critical violations|:white_check_mark: OK|0|> 0|
|Major violations|:exclamation: Error|4|> 0|
|Security hotspots reviewed|:exclamation: Error|0|< 100|
|Vulnerabilities|:white_check_mark: OK|0|> 0|

[View on SonarQube](http://sonarqube:9000/dashboard?id=pr-896)
###### _updated: 6/17/2024, 02:58:39 (UTC+0)_

**Comment by zxkane** [comment]
### SonarQube Quality Gate Result
- **Result**: :exclamation: Error
- Triggered by @zxkane on `pull_request`

| Metric | Status | Value | Error Threshold |
|:------:|:------:|:-----:|:---------------:|
|Reliability rating|:white_check_mark: OK|1|> 1|
|Security rating|:white_check_mark: OK|1|> 1|
|Sqale rating|:white_check_mark: OK|1|> 1|
|Coverage|:white_check_mark: OK|83.70|< 80|
|Duplicated lines density|:white_check_mark: OK|5.20|> 30|
|Blocker violations|:white_check_mark: OK|0|> 0|
|Bugs|:white_check_mark: OK|0|> 0|
|Code smells|:white_check_mark: OK|13|> 40|
|Critical violations|:white_check_mark: OK|0|> 0|
|Major violations|:exclamation: Error|4|> 0|
|Security hotspots reviewed|:exclamation: Error|0|< 100|
|Vulnerabilities|:white_check_mark: OK|0|> 0|

[View on SonarQube](http://sonarqube:9000/dashboard?id=pr-896)
###### _updated: 6/17/2024, 03:28:30 (UTC+0)_

**Comment by zxkane** [comment]
### SonarQube Quality Gate Result
- **Result**: :exclamation: Error
- Triggered by @dengmingtong on `pull_request`

| Metric | Status | Value | Error Threshold |
|:------:|:------:|:-----:|:---------------:|
|Reliability rating|:white_check_mark: OK|1|> 1|
|Security rating|:white_check_mark: OK|1|> 1|
|Sqale rating|:white_check_mark: OK|1|> 1|
|Coverage|:white_check_mark: OK|83.80|< 80|
|Duplicated lines density|:white_check_mark: OK|5.20|> 30|
|Blocker violations|:white_check_mark: OK|0|> 0|
|Bugs|:white_check_mark: OK|0|> 0|
|Code smells|:white_check_mark: OK|14|> 40|
|Critical violations|:white_check_mark: OK|0|> 0|
|Major violations|:exclamation: Error|5|> 0|
|Security hotspots reviewed|:exclamation: Error|0|< 100|
|Vulnerabilities|:white_check_mark: OK|0|> 0|

[View on SonarQube](http://sonarqube:9000/dashboard?id=pr-896)
###### _updated: 6/19/2024, 03:01:28 (UTC+0)_

**Comment by zxkane** [comment]
### SonarQube Quality Gate Result
- **Result**: :exclamation: Error
- Triggered by @dengmingtong on `pull_request`

| Metric | Status | Value | Error Threshold |
|:------:|:------:|:-----:|:---------------:|
|Reliability rating|:white_check_mark: OK|1|> 1|
|Security rating|:white_check_mark: OK|1|> 1|
|Sqale rating|:white_check_mark: OK|1|> 1|
|Coverage|:white_check_mark: OK|83.80|< 80|
|Duplicated lines density|:white_check_mark: OK|5.30|> 30|
|Blocker violations|:white_check_mark: OK|0|> 0|
|Bugs|:white_check_mark: OK|0|> 0|
|Code smells|:white_check_mark: OK|10|> 40|
|Critical violations|:white_check_mark: OK|0|> 0|
|Major violations|:exclamation: Error|1|> 0|
|Security hotspots reviewed|:exclamation: Error|0|< 100|
|Vulnerabilities|:white_check_mark: OK|0|> 0|

[View on SonarQube](http://sonarqube:9000/dashboard?id=pr-896)
###### _updated: 6/19/2024, 03:34:44 (UTC+0)_

File: src/ingestion-server/custom-resource/update-alb-rules/index.ts [hunk 8]
```diff
@@ -55,16 +57,19 @@ interface HandleUpdateInput {
   readonly protocol: string;
   readonly endpointPath: string;
   readonly hostHeader: string;
+  readonly enableAuthentication: string;
```
Comment (zxkane) [inline src/ingestion-server/custom-resource/update-alb-rules/index.ts:60]: we might refactor it as 

```ts
interface Input {
readonly arn: string;
readonly new: PropType;
readonly old: PropType;
}
```

File: src/ingestion-server/custom-resource/update-alb-rules/index.ts [hunk 9]
```diff
@@ -137,13 +156,14 @@ async function handleCreate(
...
+  enableAuthentication: string,
   authenticationSecretArn: string,
   targetGroupArn: string,
 ) {
   // Create defalut forward rule and action
-  await createDefaultForwardRule(listenerArn, protocol, endpointPath, domainName, authenticationSecretArn, targetGroupArn);
+  await createDefaultForwardRule(listenerArn, protocol, endpointPath, domainName, enableAuthentication, authenticationSecretArn, targetGroupArn);
 
-  if (authenticationSecretArn && authenticationSecretArn.length > 0) {
+  if (enableAuthentication === 'Yes') {
```
Comment (zxkane) [inline src/ingestion-server/custom-resource/update-alb-rules/index.ts:166]: As in the previous comment, create an enum type for the value Yes and No.

File: src/ingestion-server/custom-resource/update-alb-rules/index.ts [hunk 10]
```diff
@@ -162,8 +182,22 @@ async function handleUpdate(
...
   }
 
-  if (inputPros.authenticationSecretArn !== inputPros.oldAuthenticationSecretArn) {
-    await updateAuthenticationSecretArn(inputPros.listenerArn, inputPros.authenticationSecretArn);
+  // update authentication
+  if (
+    inputPros.enableAuthentication !== inputPros.oldEnableAuthentication ||
+    (inputPros.authenticationSecretArn !== inputPros.oldAuthenticationSecretArn && inputPros.enableAuthentication === 'Yes')
+  ) {
+    await sleep(20000);
```
Comment (zxkane) [inline src/ingestion-server/custom-resource/update-alb-rules/index.ts:190]: Why are we using a fixed value here?

File: src/ingestion-server/server-v2/ingestion-server-v2.ts [hunk 11]
```diff
@@ -228,69 +172,15 @@ export class IngestionServerV2 extends Construct {
...
-    targetGroupArn,
-    listenerArn,
-    authenticationSecretArn,
-    endpointPath,
-    domainName,
-    protocol,
-  });
-}
-
 function deleteECSCluster(scope: Construct, ecsClusterArn: string, ecsClusterName: string, ecsServiceName: string) {
```
Comment (zxkane) [inline src/ingestion-server/server-v2/ingestion-server-v2.ts:184]: change to `createDeleteECSClusterHook` is better

File: src/ingestion-server/server-v2/private/ecs-ec2-service.ts [hunk 12]
```diff
@@ -0,0 +1,232 @@
...
+  PropagatedTagSource,
+  CfnClusterCapacityProviderAssociations,
+} from 'aws-cdk-lib/aws-ecs';
+import { LogGroup, RetentionDays } from 'aws-cdk-lib/aws-logs';
+import { Construct, IConstruct } from 'constructs';
+import { ECSClusterProps, EcsServiceResult } from './ecs-ec2-cluster';
+import { addCfnNagSuppressRules, ruleToSuppressCloudWatchLogEncryption } from '../../../common/cfn-nag';
+import { DefaultEc2FleetProps, RESOURCE_ID_PREFIX } from '../ingestion-server-v2';
+
+export interface Ec2ServiceProps extends ECSClusterProps {
```
Comment (zxkane) [inline src/ingestion-server/server-v2/private/ecs-ec2-service.ts:33]: Should we have shared an abstract class for sharing the common configuration?

File: src/ingestion-server/server/parameter.ts [hunk 13]
```diff
@@ -37,6 +37,19 @@ export function createStackParameters(scope: Construct, props: {deliverToKinesis
     minValue: 60,
   });
 
+  const notificationsTopicArnParam = new CfnParameter(
```
Comment (zxkane) [inline src/ingestion-server/server/parameter.ts:40]: I think it's useless. Could we remove it?

**Review by zxkane — CHANGES_REQUESTED** [review]

**Comment by amliuyong** [comment]
image

**Comment by zxkane** [comment]
### SonarQube Quality Gate Result
- **Result**: :exclamation: Error
- Triggered by @dengmingtong on `pull_request`

| Metric | Status | Value | Error Threshold |
|:------:|:------:|:-----:|:---------------:|
|Reliability rating|:white_check_mark: OK|1|> 1|
|Security rating|:white_check_mark: OK|1|> 1|
|Sqale rating|:white_check_mark: OK|1|> 1|
|Coverage|:white_check_mark: OK|83.80|< 80|
|Duplicated lines density|:white_check_mark: OK|5.10|> 30|
|Blocker violations|:white_check_mark: OK|0|> 0|
|Bugs|:white_check_mark: OK|0|> 0|
|Code smells|:white_check_mark: OK|11|> 40|
|Critical violations|:white_check_mark: OK|0|> 0|
|Major violations|:exclamation: Error|2|> 0|
|Security hotspots reviewed|:exclamation: Error|0|< 100|
|Vulnerabilities|:white_check_mark: OK|0|> 0|

[View on SonarQube](http://sonarqube:9000/dashboard?id=pr-896)
###### _updated: 6/21/2024, 03:41:28 (UTC+0)_

**Comment by zxkane** [comment]
### SonarQube Quality Gate Result
- **Result**: :exclamation: Error
- Triggered by @zxkane on `pull_request`

| Metric | Status | Value | Error Threshold |
|:------:|:------:|:-----:|:---------------:|
|Reliability rating|:white_check_mark: OK|1|> 1|
|Security rating|:white_check_mark: OK|1|> 1|
|Sqale rating|:white_check_mark: OK|1|> 1|
|Coverage|:white_check_mark: OK|83.80|< 80|
|Duplicated lines density|:white_check_mark: OK|5.10|> 30|
|Blocker violations|:white_check_mark: OK|0|> 0|
|Bugs|:white_check_mark: OK|0|> 0|
|Code smells|:white_check_mark: OK|11|> 40|
|Critical violations|:white_check_mark: OK|0|> 0|
|Major violations|:exclamation: Error|2|> 0|
|Security hotspots reviewed|:exclamation: Error|0|< 100|
|Vulnerabilities|:white_check_mark: OK|0|> 0|

[View on SonarQube](http://sonarqube:9000/dashboard?id=pr-896)
###### _updated: 6/21/2024, 04:12:53 (UTC+0)_

**Comment by zxkane** [comment]
### SonarQube Quality Gate Result
- **Result**: :exclamation: Error
- Triggered by @dengmingtong on `pull_request`

| Metric | Status | Value | Error Threshold |
|:------:|:------:|:-----:|:---------------:|
|Reliability rating|:white_check_mark: OK|1|> 1|
|Security rating|:white_check_mark: OK|1|> 1|
|Sqale rating|:white_check_mark: OK|1|> 1|
|Coverage|:white_check_mark: OK|83.80|< 80|
|Duplicated lines density|:white_check_mark: OK|5.10|> 30|
|Blocker violations|:white_check_mark: OK|0|> 0|
|Bugs|:white_check_mark: OK|0|> 0|
|Code smells|:white_check_mark: OK|10|> 40|
|Critical violations|:white_check_mark: OK|0|> 0|
|Major violations|:exclamation: Error|1|> 0|
|Security hotspots reviewed|:exclamation: Error|0|< 100|
|Vulnerabilities|:white_check_mark: OK|0|> 0|

[View on SonarQube](http://sonarqube:9000/dashboard?id=pr-896)
###### _updated: 6/21/2024, 05:09:52 (UTC+0)_

**Comment by zxkane** [comment]
### SonarQube Quality Gate Result
- **Result**: :exclamation: Error
- Triggered by @zxkane on `pull_request`

| Metric | Status | Value | Error Threshold |
|:------:|:------:|:-----:|:---------------:|
|Reliability rating|:white_check_mark: OK|1|> 1|
|Security rating|:white_check_mark: OK|1|> 1|
|Sqale rating|:white_check_mark: OK|1|> 1|
|Coverage|:white_check_mark: OK|83.80|< 80|
|Duplicated lines density|:white_check_mark: OK|5.10|> 30|
|Blocker violations|:white_check_mark: OK|0|> 0|
|Bugs|:white_check_mark: OK|0|> 0|
|Code smells|:white_check_mark: OK|10|> 40|
|Critical violations|:white_check_mark: OK|0|> 0|
|Major violations|:exclamation: Error|1|> 0|
|Security hotspots reviewed|:exclamation: Error|0|< 100|
|Vulnerabilities|:white_check_mark: OK|0|> 0|

[View on SonarQube](http://sonarqube:9000/dashboard?id=pr-896)
###### _updated: 6/21/2024, 05:39:28 (UTC+0)_

**Comment by zxkane** [comment]
### SonarQube Quality Gate Result
- **Result**: :exclamation: Error
- Triggered by @dengmingtong on `pull_request`

| Metric | Status | Value | Error Threshold |
|:------:|:------:|:-----:|:---------------:|
|Reliability rating|:white_check_mark: OK|1|> 1|
|Security rating|:white_check_mark: OK|1|> 1|
|Sqale rating|:white_check_mark: OK|1|> 1|
|Coverage|:white_check_mark: OK|83.80|< 80|
|Duplicated lines density|:white_check_mark: OK|5.10|> 30|
|Blocker violations|:white_check_mark: OK|0|> 0|
|Bugs|:white_check_mark: OK|0|> 0|
|Code smells|:white_check_mark: OK|9|> 40|
|Critical violations|:white_check_mark: OK|0|> 0|
|Major violations|:white_check_mark: OK|0|> 0|
|Security hotspots reviewed|:exclamation: Error|0|< 100|
|Vulnerabilities|:white_check_mark: OK|0|> 0|

[View on SonarQube](http://sonarqube:9000/dashboard?id=pr-896)
###### _updated: 6/21/2024, 07:36:55 (UTC+0)_

**Comment by zxkane** [comment]
### SonarQube Quality Gate Result
- **Result**: :exclamation: Error
- Triggered by @zxkane on `pull_request`

| Metric | Status | Value | Error Threshold |
|:------:|:------:|:-----:|:---------------:|
|Reliability rating|:white_check_mark: OK|1|> 1|
|Security rating|:white_check_mark: OK|1|> 1|
|Sqale rating|:white_check_mark: OK|1|> 1|
|Coverage|:white_check_mark: OK|83.80|< 80|
|Duplicated lines density|:white_check_mark: OK|5.10|> 30|
|Blocker violations|:white_check_mark: OK|0|> 0|
|Bugs|:white_check_mark: OK|0|> 0|
|Code smells|:white_check_mark: OK|9|> 40|
|Critical violations|:white_check_mark: OK|0|> 0|
|Major violations|:white_check_mark: OK|0|> 0|
|Security hotspots reviewed|:exclamation: Error|0|< 100|
|Vulnerabilities|:white_check_mark: OK|0|> 0|

[View on SonarQube](http://sonarqube:9000/dashboard?id=pr-896)
###### _updated: 6/21/2024, 08:05:17 (UTC+0)_

**Comment by amliuyong** [comment]
image

**Comment by zxkane** [comment]
### SonarQube Quality Gate Result
- **Result**: :exclamation: Error
- Triggered by @zxkane on `pull_request`

| Metric | Status | Value | Error Threshold |
|:------:|:------:|:-----:|:---------------:|
|Reliability rating|:white_check_mark: OK|1|> 1|
|Security rating|:white_check_mark: OK|1|> 1|
|Sqale rating|:white_check_mark: OK|1|> 1|
|Coverage|:white_check_mark: OK|83.80|< 80|
|Duplicated lines density|:white_check_mark: OK|5.10|> 30|
|Blocker violations|:white_check_mark: OK|0|> 0|
|Bugs|:white_check_mark: OK|0|> 0|
|Code smells|:white_check_mark: OK|9|> 40|
|Critical violations|:white_check_mark: OK|0|> 0|
|Major violations|:white_check_mark: OK|0|> 0|
|Security hotspots reviewed|:exclamation: Error|0|< 100|
|Vulnerabilities|:white_check_mark: OK|0|> 0|

[View on SonarQube](http://sonarqube:9000/dashboard?id=pr-896)
###### _updated: 6/21/2024, 11:09:04 (UTC+0)_

**Comment by zxkane** [comment]
### SonarQube Quality Gate Result
- **Result**: :white_check_mark: OK
- Triggered by @dengmingtong on `pull_request`

| Metric | Status | Value | Error Threshold |
|:------:|:------:|:-----:|:---------------:|
|Reliability rating|:white_check_mark: OK|1|> 1|
|Security rating|:white_check_mark: OK|1|> 1|
|Sqale rating|:white_check_mark: OK|1|> 1|
|Coverage|:white_check_mark: OK|83.80|< 80|
|Duplicated lines density|:white_check_mark: OK|5.10|> 30|
|Blocker violations|:white_check_mark: OK|0|> 0|
|Bugs|:white_check_mark: OK|0|> 0|
|Code smells|:white_check_mark: OK|9|> 40|
|Critical violations|:white_check_mark: OK|0|> 0|
|Major violations|:white_check_mark: OK|0|> 0|
|Vulnerabilities|:white_check_mark: OK|0|> 0|

[View on SonarQube](http://sonarqube:9000/dashboard?id=pr-896)
###### _updated: 6/21/2024, 13:19:07 (UTC+0)_

File: src/ingestion-server/custom-resource/update-alb-rules.ts [hunk 14]
```diff
@@ -27,6 +27,7 @@ export interface UpdateAlbRulesCustomResourceProps {
   clickStreamSDK: string;
   targetGroupArn: string;
   listenerArn: string;
+  enableAuthentication: string;
```
Comment (zxkane) [inline src/ingestion-server/custom-resource/update-alb-rules.ts:30]: use `FeatureFlag`

**Review by zxkane — CHANGES_REQUESTED** [review]

**Review by zxkane — APPROVED** [review]

**Pull request merged**